autoload/neomake/makers/ft/elixir.vim

```diff
@@ -1,12 +1,28 @@
 " vim: ts=4 sw=4 et
 
+let g:neomake_elixir_credo_type_map= get(g:, 'neomake_elixir_credo_type_map', {'F': 'W','C': 'W','D': 'I','R': 'I'})
+
 function! neomake#makers#ft#elixir#PostprocessEnforceMaxBufferLine(entry) abort
     let buffer_lines = str2nr(line('$'))
     if (buffer_lines < a:entry.lnum)
         let a:entry.lnum = buffer_lines
     endif
 endfunction
 
+" Credo error types
+" F -> Refactoring opportunities
+" W -> Warnings
+" C -> Convention violation
+" D -> Software design suggestions
+" R -> Readability suggestions
+function! neomake#makers#ft#elixir#PostprocessCredoErrorType(entry) abort
+    let type = toupper(a:entry.type)
+    let type_map = g:neomake_elixir_credo_type_map
+    if has_key(type_map, type)
+        let a:entry.type = type_map[type]
+    endif
```

Comment:
Regarding tests: there are some `tests/ft_*` cases already, in case you want to dig into it.